tracing-subscriber/src/fmt/format/mod.rs

```diff
@@ -806,6 +809,24 @@ impl<'a> fmt::Debug for DefaultVisitor<'a> {
     }
 }
 
+/// Renders an error into a list of sources, *including* the error
```

Comment:
do we really want to include the error? it seems weird to format the error once in its own field *and* a second time in the list of its sources...is an error really its own source? 

including the error here will result in output like
```
error=err1 error.sources=[err1, err2, err3]
```

i think it would be better to do one of the following, instead:
1. just output the entire chain as a list when the error has sources, like this:
   ```
   error=[err1, err2, err3]
   ```
2. output the first error as the field, and then output only its sources (not the initial error) in the field list, like
   ```
   error=err1 error.sources=[err2, err3]
   ```

what do you think?

Reply:
Both formatters only give `err1.source()` to `ErrorSourceList` anyway, since they need to make the decision about whether to display `error.sources` at all, so the PR currently implements approach 2.

As for the question of which approach to prefer, IMO 1 makes sense for machine-readable formats (to avoid addingg special cases for parsers) while 2 is preferable for human-readable output (since it puts the empasis on the leaf error that explains the symptoms).

Reply:
> Both formatters only give `err1.source()` to `ErrorSourceList` anyway, since they need to make the decision about whether to display `error.sources` at all, so the PR currently implements approach 2.

oh, lol, I totally misinterpreted this code --- I thought `ErrorSourceList` was being constructed with the initial error. Never mind!